src/test/kotlin/org/rust/ide/inspections/lints/RsNeedlessLifetimesInspectionTest.kt

```diff
@@ -355,6 +355,14 @@ class RsNeedlessLifetimesInspectionTest : RsInspectionsTestBase(RsNeedlessLifeti
         fn foo2<'a>(_: &'a str) {}
     """)
 
+    fun `test async fn`() = checkByText("""
+        struct S<'a>(&'a str);
+
+        async <weak_warning>fn foo1<'a>(a: S<'a>) -> S<'a></weak_warning> {}
+
+        async fn foo2<'a>(a: &'a str) -> &'a str {}
```

Comment:
When I elide the lifetime from `foo2` signature, the code still compiles fine. So maybe the logic is a bit different and the inspection should mark this one with a weak warning too.